include/envoy/upstream/upstream.h

```diff
@@ -313,6 +313,11 @@ class ClusterInfo {
   virtual LoadBalancerType lbType() const PURE;
 
   /**
+   * @return configuration for ring hash load balancing, only used if type is set to ring_hash_lb.
+   */
+  virtual const envoy::api::v2::Cluster::RingHashLbConfig lbRingHashConfig() const PURE;
```

Comment:
To mirror the dataplane-api code, should this return something that corresponds to the oneof lb_config?  I'm not sure what the right way pass a "base class" of a protobuf message is, that can be downcast to the correct type.

If not, this should probably be Optional<>.

Reply:
Probably need Optional<>, yeah. I was initially viewing the oneof as input enforcement in the proto and not extending to the code itself.

Let's see what @htuch has to say.

Reply:
At the interface level here I would probably just do Optional<>, or return a const pointer which can be nullptr. Either way.